### DISCUSSION TITLE
Docker Auth Manager: How gets a Private registry containing a port gets split?

### DISCUSSION THREAD

**Discussion by steffenboe** [renovatebot/renovate#8200] [Request Help] [answered]
Hi everyone,
i have a private docker registry containing a port in its baseUrl, which i try to authenticate against. Now the strange thing is that i get the message
``DEBUG: Failed to get authHeaders for getTags lookup ``

When i curl the same docker registry baseUrl (like ``curl -i https://myDockerRegistry:5000/v2/``) i get auth-headers.

I found the file ``/lib/datasource/index.ts`` which handles the authentication process for docker registries. There is a function called ``getRegistryRepository`` handling the splitting of registry/repository: https://github.com/renovatebot/renovate/blob/master/lib/datasource/docker/index.ts#L61. 
So as i am not familiar with typescript: does the logic in ``getRegistryRepository`` considers ports in docker registry baseurls?

**Comment by steffenboe** [comment, marked as answer]
solved this: when setting the property `domainName` to something like ``com.something.myregistry:5000``, it failed to get headers. setting instead `baseUrl` to ``https://com.something.myregistry:5000/`` everything works fine. so domainName does apparently not support ports (which makes sense... kind of)

**Reply by rarkins** [reply]
Domain names and host names cannot contain ports, so it makes complete sense. I have created #8211 for us to validate them in future to catch this type of config error.
